app/jobs/regular/process_post.rb

```diff
@@ -60,6 +60,26 @@ def extract_links(post)
       TopicLink.extract_from(post)
       QuotedPost.extract_from(post)
     end
+
+    def auto_tag(post)
+      topic_tags = post.topic.tags.pluck(:name)
+      changed = false
+
+      WordWatcher.words_for_action(:tag).each do |word, tags|
+        if SiteSetting.watched_words_regular_expressions? ? Regexp.new(word).match?(post.raw) : post.raw.include?(word)
```

Comment:
Is that the only place we're doing the matching? Feels like this logic should be hidden behind a method to which you would pass the `word` and the `string` (aka. `post.raw`) and it would automatically do the matching based on the value of the `watched_words_regular_expressions` site setting

Reply:
We do this matching in another place, but that is different because for tags we have to know which tags are "assigned" to a word. It is not enough that "word" matches, we also need the "replacement" value assigned to that word.